File: students/students/src/MON.Models/BasicDocumentPartModel.cs
namespace MON.Models
{
    using System.Collections.Generic;

    public class BasicDocumentPartUpdateModel
    {
        public int? Id { get; set; }
        public int Position { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Code { get; set; }
        public bool? IsHorariumHidden { get; set; }
        public int? BasicClassId { get; set; }
        public int? PrintedLines { get; set; }
        public int? TotalLines { get; set; }
        public List<int> SubjectTypes { get; set; } = new List<int>();
        public List<int> ExternalEvaluationTypes { get; set; }
        public List<BasicDocumentSubjectModel> BasicDocumentSubjects { get; set; }
    }
}

File: students/students/src/MON.SPA/src/models/diploma/basicDocumentPartModel.js
export class BasicDocumentPartModel {
  constructor(obj = {}) {
    this.uid = obj.uid || Math.floor((Math.random()) * 0x10000).toString(16);
    this.id = obj.id;
    this.position = obj.position || 1;
    this.name = obj.name;
    this.description = obj.description;
    this.code = obj.code;
    this.isHorariumHidden = obj.isHorariumHidden;
    this.basicClassId = obj.basicClassId;
    this.subjectTypes = obj.subjectTypes;
    this.externalEvaluationTypes = obj.externalEvaluationTypes;
    this.basicDocumentSubjects = obj.basicDocumentSubjects;
    this.printedLines = obj.printedLines;
    this.totalLines = obj.totalLines;
  }
}
